chips/nrf51/src/trng.rs

```diff
@@ -0,0 +1,132 @@
+//! TRNG driver for the nrf51dk
+//! Author: Niklas Adolfsson <niklasadolfsson1@gmail.com>
+//! Author: Fredrik Nilsson <frednils@student.chalmers.se>
+//! Date: March 01, 2017
+
+use chip;
+use core::cell::Cell;
+use core::mem;
+use kernel::hil::rng::{self, Continue};
+use nvic;
+use peripheral_interrupts::NvicIdx;
+use peripheral_registers::{RNG_BASE, RNG_REGS};
+
+pub static mut DMY: [u8; 4] = [0; 4];
+
+pub struct Trng<'a> {
+    regs: *mut RNG_REGS,
+    client: Cell<Option<&'a rng::Client>>,
+    done: Cell<u8>,
+}
+
+pub static mut TRNG: Trng<'static> = Trng::new();
+
+impl<'a> Trng<'a> {
+    const fn new() -> Trng<'a> {
+        Trng {
+            regs: RNG_BASE as *mut RNG_REGS,
+            client: Cell::new(None),
+            done: Cell::new(0),
+        }
+    }
+
+    // ONLY VALRDY CAN TRIGGER THIS INTERRUPT
+    pub fn handle_interrupt(&self) {
+        let regs: &mut RNG_REGS = unsafe { mem::transmute(self.regs) };
+
+        // disable interrupts
+        self.disable_interrupts();
+        self.disable_nvic();
+        regs.STOP.set(1);
+        nvic::clear_pending(NvicIdx::RNG);
+
+        match self.done.get() {
```

Comment:
Please inline document these branches to make it a bit easier to follow the logic.